rest-service/manager_rest/rest/resources.py

```diff
@@ -803,13 +822,278 @@ def get(self, deployment_id, **kwargs):
 
 class Events(SecuredResource):
 
+    """Events resource.
+
+    Throgh the events endpoint a user can retrieve both events and logs as
+    stored in the SQL database.
+
+    """
+
     @staticmethod
-    def _query_events():
+    def _build_select_query(_include, filters, pagination, sort):
+        """Build query used to list events for a given execution.
+
+        :param _include:
+            Projection used to get records from database (not currently used)
+        :type _include: list(str)
+        :param filters:
+            Filter selection. It's used to decide if events:
+                {'type': ['cloudify_event']}
+            or both events and logs should be returned:
+                {'type': ['cloudify_event', 'cloudify_log']}
+        :type filters: dict(str, str)
+        :param pagination:
+            Parameters used to limit results returned in a single query.
+            Expected values `size` and `offset` are mapped into SQL as `LIMIT`
+            and `OFFSET`.
+        :type pagination: dict(str, int)
+        :param sort:
+            Result sorting order. The only allowed and expected value is to
+            sort by timestamp in ascending order:
+                {'timestamp': 'asc'}
+        :type sort: dict(str, str)
+        :returns:
+            A SQL query that returns the events found that match the conditions
+            passed as arguments.
+        :rtype: :class:`sqlalchemy.sql.elements.TextClause`
+
         """
-        List events for the provided Elasticsearch query
+        if _include is not None:
+            current_app.logger.error(
+                'Projections with `_include` parameter are not supported')
+            return abort(400)
```

Comment:
In other request handlers a BadParametersError is raised which is then handled by the "exceptions_handled" decorator.. this is done in order to return a consistent JSON response in case there's an error with some known error codes. There are many examples for this in the code, just look for BadParametersError.